pkg/server/endpoints/registration/handler.go

```diff
@@ -52,35 +52,50 @@ func (h *Handler) CreateEntry(ctx context.Context, request *common.RegistrationE
 	telemetry_common.AddCallerID(counter, getCallerID(ctx))
 	log := h.Log.WithField(telemetry.Method, telemetry.CreateRegistrationEntry)
 
-	request, err = h.prepareRegistrationEntry(request, false)
-	if err != nil {
-		err = status.Error(codes.InvalidArgument, err.Error())
-		log.WithError(err).Error("Request parameter validation error")
-		return nil, err
-	}
-
-	ds := h.getDataStore()
-
-	unique, err := h.isEntryUnique(ctx, ds, request)
-	if err != nil {
-		log.WithError(err).Error("Error trying to create entry")
-		return nil, status.Errorf(codes.Internal, "error trying to create entry: %v", err)
+	entry, preexisting, gErr := h.createRegistrationEntry(ctx, request)
+	if gErr != nil {
+		switch gErr.Code() {
+		case codes.InvalidArgument:
+			log.WithError(gErr.Err()).Error("Request parameter validation error")
+			return nil, gErr.Err()
+		default:
+			log.WithError(gErr.Err()).Error("Error trying to create entry")
+			return nil, gErr.Err()
+		}
```

Comment:
I think it looks a little cleaner if `createRegistrationEntry` returned an error instead of a status...

```suggestion
	entry, preexisting, err := h.createRegistrationEntry(ctx, request)
	if err != nil {
		switch status.Code(err) {
		case codes.InvalidArgument:
			log.WithError(err).Error("Request parameter validation error")
		default:
			log.WithError(err).Error("Error trying to create entry")
		}
		return nil, err
```

Reply:
Yes, that is much better. Updated!